service/src/main/kotlin/xyz/lebkuchenfm/external/storage/mongo/repositories/XSoundsMongoRepository.kt

```diff
@@ -51,6 +52,10 @@ class XSoundsMongoRepository(database: MongoDatabase) : XSoundsRepository {
         ).first()
         return result.tagsSet
     }
+
+    override suspend fun findByName(name: String): XSound? {
+        return collection.find(eq(XSound::name.name, name)).firstOrNull()?.toDomain()
```

Comment:
`XSoundEntity::name.name, name` - we should use mongo's entity here
Please fix also filter in line 32 🙏🏻